src/main/peripherals/Lock.java

```diff
@@ -0,0 +1,46 @@
+package peripherals;
+import java.io.IOException;
+import java.net.InetAddress;
+
+
+
+public class Lock implements IPeripheral{	
+	private String ip;
+	private String port;
+	
+	public Lock(String ip, String port){
+		this.ip = ip;
+		this.port = port;
+	}
+	
+
+	public Lock(InetAddress ip, int port){
+		this.ip = ip.toString();
+		this.port = Integer.toString(port);
+	}
+	
+	@Override
+	public void performAction(String action) throws ActionNotDefinedException, IOException, InterruptedException{
+		if(action.equals("openLock")){
```

Comment:
A command pattern might be better here, then you could move the `sendCurl` method into a generic command as well.